community/kernel/src/main/java/org/neo4j/kernel/impl/traversal/BidirectionalTraversalDescriptionImpl.java

```diff
@@ -143,9 +143,20 @@ public Traverser traverse( Node start, Node end )
     @Override
     public Traverser traverse( final Iterable<Node> startNodes, final Iterable<Node> endNodes )
     {
-        return new DefaultTraverser( () -> new BidirectionalTraverserIterator(
-                statementFactory.get(), start, end, sideSelector, collisionPolicy, collisionEvaluator, maxDepth,
-                startNodes, endNodes ) );
+        return new DefaultTraverser( () ->
+        {
+            Resource resource = statementFactory.get();
+            try
+            {
+                return new BidirectionalTraverserIterator( resource, start, end, sideSelector, collisionPolicy,
+                        collisionEvaluator, maxDepth, startNodes, endNodes );
+            }
+            catch ( Exception e )
```

Comment:
`try-finally` will not help here since its exceptions that `BidirectionalTraverserIterator` can throw during instantiation we covering from here, in normal case this iterator will be closed later